third_party/amd/backend/compiler.py

```diff
@@ -296,6 +296,15 @@ def make_llir(src, metadata, options):
         passes.convert.add_index_to_llvmir(pm)
 
         amd.passes.ttgpuir.add_allocate_shared_memory(pm)
+
+        # Perform the membar analysis before the ops refinement (the refinement pass is coming later)
+        # to prevent redundant `gpu.barriers` from ops on sub-sliced `ttg.memdesc_subviews`.
```

Comment:
sorry for repeating myself, mentioned it already in the other thread but I still don't think this is a good direction. Maybe we should have a simple place to discuss this